CauldronMods/Controller/Heroes/MagnificentMara/Cards/AbracadabraCardController.cs

```diff
@@ -48,6 +49,7 @@ private IEnumerator MayReturnDestroyedResponse(DestroyCardAction dc)
 
             if (DidPlayerAnswerYes(decisionResult))
             {
+                Journal.RecordCardProperties(Card, SavedCard, dc.CardToDestroy.Card);
                 dc.SetPostDestroyDestination(dc.CardToDestroy.Card.Owner.ToHero().Hand);
                 dc.AddAfterDestroyedAction(() => DestroyThisCardIfMovedResponse(dc), this);
```

Comment:
Just had a thought: Does `dc.AddAfterDestroyedAction(() => DestroyThisCardIfMovedResponse(dc, this), this)` and modifying DestroyThisCardIfMovedResponse to take a cardcontroller to destroy work?

Reply:
I have tried that, both as an instance method and an extension method of `CardController`. Neither worked.

Reply:
Just did some fiddling; `dc.AddAfterDestroyedAction(() => DestroyThisCardIfMovedResponse(dc, Card), this);` and `private IEnumerator DestroyThisCardIfMovedResponse(DestroyCardAction dc, Card savingCard)` passes the tests. Passing `this` was a silly idea and won't work because it's just the CardController, but the `CardController.Card` property does the appropriate replacement magic. Good plan on solving the problem and thank you for working out what's going wrong. :)